src/cache/cache_data.h

```diff
@@ -6,15 +6,15 @@
 
 #ifndef CACHE_DATA_H
 #define CACHE_DATA_H
-#include "./result_set_cache_includes.h"
+#include "./cache_includes.h"
 
 /**
  * @brief  struct for holding cache data
  */
 typedef struct CacheData
 {
-  char hashKey[HASH_KEY_LENGTH];    // CacheData key - Query hashed by XXHASH
-  ResultSet *resultSet;             // Value - ResultSet
+  unsigned long long hashKey;    // CacheData key - Query hashed by XXHASH
+  void *cacheValue;             // Value - ResultSet
```

Comment:
Specifying ResultSet here doesn't implies generic.

Reply:
don't know why you have seen that, its void*